### DISCUSSION TITLE
What is a CommandMapper?

### DISCUSSION THREAD

**Discussion by mcguu** [dotnet/maui#5230] [Q&A]
Is it a command to the view?

**Comment by hartez** [comment]
CommandMappers provide a way for cross-platform views to send commands to the concrete views on each platform. They're very similar to PropertyMappers, but CommandMappers allow for extra command parameters to be passed.

A good example of a CommandMapper use is the ScrollView's ScrollTo command. The cross-platform ScrollView asks its handler to invoke the ScrollTo command, and passes along the scroll arguments (the position or element it wants to scroll to). The ScrollView's handler on each platform knows how to unpack the arguments and run that command with the concrete platform view (e.g., a UIScrollView on iOS, or a ScrollViewer on Windows). 

In Forms, this sort of thing was done with an event on the cross-platform view, and the renderer would subscribe to the event. The CommandMapper approach decouples the concrete view, and avoids some of the pitfalls of events. It also allows for easy customization; if you want to inject code into ScrollTo handling or replace it completely, you can simply modify the mapping in the CommandMapper - no subclassing or renderer customization required.

**Reply by hartez** [reply]
In a case like that, the handler would be responsible for subscribing to the the event from the concrete view, and in the even handler it would Invoke the cross-platform view's command. (If I understood your question correctly.)

**Reply by mcguu** [reply]
Thank you I wonder if there is any case someone to teach the code of the maui each function and each feature.

**Reply by jsuarezruiz** [reply]
You can read more about Handlers and the Mapper here https://github.com/jsuarezruiz/xamarin-forms-to-net-maui/tree/main/Handlers
I need to update it to include details about the CommandMapper.

**Reply by mcguu** [reply]
when will you use UpdatePlaceholderColor?If you change the color you have to do it in the android?

entry1.PlaceHolderColor=Color.FromArgb("#000000");

then the color not changed? and you have to do it in the android renderer?

**Reply by F-Goncalves** [reply]
Still no details on how the CommandMapper is supposed to work anywhere? Please include it in the documentation.

**Reply by marcojak** [reply]
Looking for some documentation for the CommandMapper. Not there yet.

**Reply by dotMorten** [reply]
@jsuarezruiz Any update on adding CommandMapper to this sample? The doc and samples on this could still use a bit of work

**Reply by jsuarezruiz** [reply]
Right, I don't include CommandMapper in that example, etc. I'm going to add it but let me also include in this thread @davidbritch to take this into account for the official documentation.
